src/renderer/views/stake/StakeView.tsx

```diff
@@ -12,28 +14,44 @@ import { useParams } from 'react-router-dom'
 import { AddStakeStory } from '../../components/stake/AddStake/AddStake.stories'
 import { Stake } from '../../components/stake/Stake'
 import BackLink from '../../components/uielements/backLink'
-/**
- * temporary data mock
- * @TODO @thatStrangeGuyThorchain replace mocck after
- * https://github.com/thorchain/asgardex-electron/issues/441 resolved
- */
-import { defaultPoolShare } from '../../components/uielements/poolShare/PoolShare.stories'
+import { useBinanceContext } from '../../contexts/BinanceContext'
 import { useMidgardContext } from '../../contexts/MidgardContext'
+import { useWalletContext } from '../../contexts/WalletContext'
 import { StakeRouteParams } from '../../routes/stake'
 import { PoolDetailsView } from './PoolDetails/PoolDetailsView'
+import { ShareView } from './Share/ShareView'
 
 type Props = {}
 
 const StakeView: React.FC<Props> = (_) => {
   const { asset: assetParam } = useParams<StakeRouteParams>()
   const { service: midgardService } = useMidgardContext()
+  const { keystoreService } = useWalletContext()
+  const { address$ } = useBinanceContext()
 
-  const asset = assetFromString(assetParam)
+  const keystore = useObservableState(keystoreService.keystore$, O.none)
 
-  useEffect(() => {
-    midgardService.pools.reloadPoolDetailedState(O.fromNullable(asset))
-  }, [asset, midgardService.pools])
+  const hasWallet = useMemo(
+    () =>
+      pipe(
+        keystore,
+        O.flatten,
+        O.map(Boolean),
+        O.getOrElse(() => false)
+      ),
+    [keystore]
+  )
+
+  const asset = assetFromString(assetParam.toUpperCase())
```

Comment:
```
const asset = O.fromNullable(assetFromString(assetParam.toUpperCase()))
```
and "jump out" to show an error view only if asset is not valid or/and not available (similar to a previous comment).

Reply:
not sure that i understand what you mean by `FIRST`

Reply:
we can not do things in this way. react registers hooks in an order they are declared and uses just an array of hooks to use appropriate hook. more details are [here](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)